### ISSUE TITLE
Ariel, after 3.3.1, does not remain running when laptop is closed. Have to restart it every time. Pretty annoying.

### ISSUE THREAD

**Issue by greylogan** [JohnCoates/Aerial#1331]
#### General troubleshooting tips

Before logging an issue please check that:
- [ ] You have the latest version installed (There may be a beta version that fixes your issue), see here for the latest releases and bug fixes: https://github.com/JohnCoates/Aerial/releases
- [ ] Your issue isn't already mentioned in our [issues](https://github.com/JohnCoates/Aerial/issues). You may find a workaround there or a similar request already made.
- [ ] Your problem isn't mentioned in the [troubleshooting page](https://github.com/JohnCoates/Aerial/blob/master/Documentation/Troubleshooting.md).

If none of this fixes your issue, tell us about the problem you are experiencing or the feature you'd like to request.

#### Required information
In order to help us sort your issue, we ask that you provide the following information:
- [ ] Mac model: 
- [ ] macOS version: 
- [ ] Monitor setup: 

If appropriate, please enable `Debug mode` and `Log to disk` in `Advanced` tab and replicate your bug, then attach the `AerialLog.txt` file (You can access this file through the Advanced tab).

#### Description of issue / Feature request

{{Replace this}}

**Comment by glouel** [comment]
Hey @greylogan 

Not sure I understand, can you expand a bit ? You close your laptop but you don't have external monitors ? What do you need to restart ? Are you talking about the desktop background or the saver ?

**Comment by greylogan** [comment]
Thank you for your response. When I close the lid and then open it later on to resume working on the laptop, Arial has closed itself and the live wallpaper under Experimental has stopped completely. I would really like it if it would continue like the previous version. The new update seems to have caused Arial to force close upon closing the lid, which closes using the screensaver as a live wallpaper as well.
________________________________

**Comment by glouel** [comment]
Ok so, this is a Sonoma thing that I may not have accounted for. However, you should not use that old version of Companion (the menu bar thing), please use this version for Sonoma, and use the desktop from there (2nd option), see if it still does it when you close the lid. I dont have a laptop to test but it's fine if my MacStudio goes to sleep ,so that should be about the same. 

https://github.com/AerialScreensaver/AerialCompanion/releases/tag/v1.5.1

**Comment by greylogan** [comment]
Hi. I installed the Sonoma version and for some reason it is doing the same thing as the 3.3.1 version. It force closes and does not stay open upon closing the laptop and opening it up again. I have to restart it every time.

**Comment by glouel** [comment]
Thanks for the feedback, I'll try to see if I can simulate something here to replicate and will keep you posted. I assume that if you let you mac fall asleep with the lid on, it's fine though.

Will let you know if I can find a workaround

**Comment by glouel** [comment]
Also, can you give me the exact steps you do ? Is the screen saver (not the desktop thing) running when you close the lid ? etc etc

Thanks, that would help me narrow it down.

**Comment by greylogan** [comment]
Basically I will close the lid and travel somewhere like home. Then I will open the Macbook and notice the icon is no longer at the top of the task bar.

Ok I just received dialogue for the first time telling me how to choose Arial as my screensaver. I never noticed this before when I clicked the button within the Arial app in the task bar. I will see if this fixes the issue. It should be set to Arial and not the default Apple screensaver, but I am not sure that will fix the issue using the screensaver as a live desktop. I will see if it stays active.

**Comment by glouel** [comment]
I see, you weren't running the screensaver but maybe Apple's one. All in all you can use anything you want and use the Companion's desktop independently, but it's really confusing right now with the way Apple presents their stuff.

What I'm asking specifically is when you close the lid. Is your mac running with apps and stuf in the foreground or was (any) screensaver running ?

**Comment by glouel** [comment]
I found a way to reproduce the issue, I'll keep you posted!

**Comment by glouel** [comment]
@greylogan found the issue, this is fixed in 3.3.2 https://github.com/JohnCoates/Aerial/releases/tag/v3.3.2

To update using Companion, if you don't see it when you open the app, you can go into settings (second icon at the bottom) and press "Check now" in the window that opens. 

After update, you have to close Companion and restart it so the new version gets loaded properly. Please let me know if that fixes things for you but I'm pretty sure it should.

**Comment by glouel** [comment]
You can't do attachements by email, you have to do them on GitHub. But I'm pretty sure it's the new sandboxing prompt saying that Aerial needs access to other applications data.

This is an unfortunate new thing that Apple added in Sonoma, the fix is simple though, see this screenshot : 

![Capture d’écran 2023-10-04 à 13 26 02](https://github.com/JohnCoates/Aerial/assets/37544189/ee7ea14e-a657-43ab-a211-87f272fab704)

You have to explicitly grant Aerial full disk access. 

This happens because I distribute 2 bits of software, the first one is Aerial (the screensaver), and the second is Aerial Companion, which is the auto updater and the part that lets you run the desktop background. In order to make the desktop background work, what Aerial Companion does is it loads Aerial (the screensaver), and that triggers this prompt.

In a perfect world, because I distribute both bits together, it shouldn't trigger anything, but because Apple still hasn't released a new SDK for screensavers, they don't, and we get this prompt. It's super annoying but I can't do much about it. 

You may have the same issues with text editors or other apps that need to open files, full disk access is the only workaround for now.

**Comment by greylogan** [comment]
Thank you. When I get home, I'll implement this fix. Yeah, Apple makes it hard to do anything.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by glouel** [comment]
After having both 3.3.2 and having restarted it, it should no longer prompt quit on your though, make sure it's properly running. You can check in settings that you have the correct version installed.

**Comment by greylogan** [comment]
OK will do. Also, I don't have 3.3.2 any longer. I deleted that and installed the Sonoma you shared a link to.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by glouel** [comment]
Let me recap because it's confusing, first you need Aerial Companion 1.5.1, this is the thing on your menu bar, see the screenshot (again, come to github to see them)  : 

![Capture d’écran 2023-10-04 à 15 02 51](https://github.com/JohnCoates/Aerial/assets/37544189/c9712f0a-7adb-4760-9be9-3bbae00ac934)

In here, if you press the cog icon top right, it will load the screen saver settings, if you go to the bottom it should show 3.3.2 (which is the screensaver version) : 

![Capture d’écran 2023-10-04 à 15 04 15](https://github.com/JohnCoates/Aerial/assets/37544189/40c47ac9-1bc5-49c2-85c6-d15b66ba145c)

If you have those things aligned, then you should no longer see the quit thing.

**Comment by greylogan** [comment]
OK. I'll check when I get home and fix the Apple settings

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by greylogan** [comment]
I was wondering what screen this is. It's in French or something I can read it.

**Comment by glouel** [comment]
Yes sorry, I happen to be french ;) Look for "Full disk access" in settings, you'll find it.

**Comment by greylogan** [comment]
Got it. And the funny thing is, before I even turned it on, I opened the lid and it was running! And I didn't get the pop up this time. So maybe it corrected itself? Either way, I granted full disk access so maybe Apple won't throw a tantrum anymore lol
